R/uvhydrograph-data.R

```diff
@@ -225,17 +225,63 @@ parseLabelSpacing <- function(data, info) {
   
   if (names(data) %in% c("series_corr", "series_corr_ref", "series_corr_up", "series_corr2")){
     limits <- info[[grep("lims_UV", names(info))]]
-    y_positions <- rep(limits$ylim[2], length(data[[1]]$time))
-    differences <- as.numeric(diff(data[[1]]$time))
-    if(length(differences) > 0) {
-      for (i in seq_len(length(differences))) {
-        if(abs(differences[i]) < 86400) {y_positions[i+1] <- y_positions[i]-(0.04*info$lims_UV$ylim[2])}
-        i <- i + 1
-      }
-      spacingInfo <- list(y=y_positions, label=seq(length(data[[1]]$time)))
-    } else {
-      spacingInfo <- list(y=y_positions, label=seq(length(data[[1]]$time)))
-    }
+
+    #The percentage of the y-range to subtract each time we add a new label to a column
+    subtractor <- (limits$ylim[[2]] - limits$ylim[[1]]) * 0.065
+
+    #Generate necessary data for corrections label spacing
```

Comment:
Yeah, this is a really huge piped dplyr call. I spent quite awhile working on the logic to try to make it shorter but couldn't come up with anything shorter than this. If you have any suggestions let me know! I tried to comment it well enough that it makes sense but I'd be happy to walk through the logic, it's certainly possible that I am doing more than necessary or that I missed something.